opm/simulators/wells/BlackoilWellModel_impl.hpp

```diff
@@ -826,7 +822,7 @@ namespace Opm {
             std::vector< Scalar > B_avg(numComponents(), Scalar() );
             computeAverageFormationFactor(B_avg);
 
-            if (param_.solve_welleq_initially_ && iterationIdx == 0) {
+            if (param_.solve_welleq_initially_) {
```

Comment:
As you write, this change should be well tested before merging. I assume it helps or is necessary for the case motivating the work. Have you tested any other complex models?